### ISSUE TITLE
Release to ROS Melodic

### ISSUE THREAD

**Issue by davetcoleman** [ros-planning/moveit#806]
A checklist of items for the next ROS 1.0 LTS Release:

- [x] branch for Melodic @davetcoleman
- Meet dependencies @130s
  - [x] object_recognition_msgs -> Requested https://github.com/wg-perception/object_recognition_msgs/issues/6
  - [x] octomap_msgs -> Requested https://github.com/OctoMap/octomap_msgs/issues/9
  - [x] moveit_msgs -> done https://github.com/ros/rosdistro/pull/17493
  - [x] warehouse_ros -> done https://github.com/ros/rosdistro/pull/17319
  - [x] ompl -> Asked https://github.com/ros-gbp/ompl-release/issues/13
  - [x] moveit_resources -> done https://github.com/ros/rosdistro/pull/17532
  - [x] vision_opencv -> asked https://github.com/ros-perception/vision_opencv/issues/208 -> Done? https://github.com/ros-perception/vision_opencv/issues/208#issuecomment-386090154
  - [x] libfcl-dev -> asked https://github.com/ros/rosdistro/pull/17765
  - [ ] graph_msgs @davetcoleman
  - [x] python-pyassimp -> asked https://github.com/ros/rosdistro/pull/17766
  - [x] OMPL release is missing cmake config @mamoll (https://github.com/ros-planning/moveit/issues/806#issuecomment-390029076)
  - [x] [Anything else](http://repositories.ros.org/status_page/blocked_releases_melodic.html?q=moveit&s=6&r=1)
- [ ] setup moveit_ci buildfarm @davetcoleman
- [x] class_loader headers [migration](http://wiki.ros.org/melodic/Migration) @jrgnicho -> done https://github.com/ros-planning/moveit/issues/806#issuecomment-389760536
- [x] pluginlib headers [migration](http://wiki.ros.org/melodic/Migration) @jrgnicho -> done https://github.com/ros-planning/moveit/issues/806#issuecomment-389760536
- [x] #901: adapt to OMPL 1.4 API
- [x] #896 (https://github.com/ros-planning/moveit/issues/806#issuecomment-389836358)
- [x] optional: [tf2 migration](https://github.com/ros-planning/moveit/pull/830) (required for ROS2) @IanTheEngineer 
- [ ] optional: [python3 migration](https://github.com/ros-infrastructure/rep/pull/149) (required for ROS2) @bmagyar?
- Cleanup, fixes, API changes:
  - [ ]  remove stuff that was indicated as deprecated in Kinetic
- [x] run bloom prebuild to check for issues @130s 

**Can be after release**
- [x] Make an announcement for "beta" release -> done https://github.com/ros-planning/moveit.ros.org/pull/167
- [ ] update README.md with new release badges
- [ ] update moveit website install instructions, remove lunar

**Comment by awesomebytes** [comment]
Hello, I could give a hand with the python3 migration and the Python interface in general with @bmagyar.

**Comment by 130s** [comment]
- Meet dependencies
  - [ ] object_recognition_msgs -> Requested https://github.com/wg-perception/object_recognition_msgs/issues/6
  - [ ] octomap_msgs -> Requested https://github.com/OctoMap/octomap_msgs/issues/9
  - [ ] moveit_msgs

**Comment by davetcoleman** [comment]
https://github.com/ros-planning/srdfdom/pull/36
Melodic development has begun! :busts_in_silhouette:

**Comment by bmagyar** [comment]
#804 should be squeezed in too

**Comment by davetcoleman** [comment]
I've begun working on the Docker / CI for Melodic task

**Comment by 130s** [comment]
- Maintainers, please hold merges (except critical ones). I'm planning to make a release soon.
- The following 2 seem to be resolved in https://github.com/ros-planning/moveit/pull/827#issuecomment-389453946, so I checked them.
  > - [ ]  class_loader headers migration @jrgnicho
  > - [ ] pluginlib headers migration @jrgnicho
- [All dependency seems to be cleared](http://repositories.ros.org/status_page/blocked_releases_melodic.html?q=moveit&s=6&r=1) so I think I'm ready to `bloom` but have a few questions:
  - There are still some check items unchecked. Are they a requirement for `bloom`? Particularly I have no idea if the following is required:
  > - Cleanup, fixes, API changes:
  >   - [ ]    remove stuff that was indicated as deprecated in Kinetic

  - Is `graph_msgs` required as it's listed above as a dependency? I do not see any mention in the code.
  ```
  $ git branch 
  * melodic-devel
  $ ack-grep -A 1 -B 1 -i graph_msgs .
  $ git log -1
  commit 6cca2555a540be43c085f2024f3cebe33fa670ea
  Author: Robert Haschke <rhaschke@users.noreply.github.com>
  Date:   Wed May 16 16:37:52 2018 +0200
  
      removing moveit_tutorials from moveit.rosinstall (#898)
      
      This partially reverts 068dcc0b82e512c7405cfa51009652516ccc9b2b, which added the tutorials just recently.
  ```

**Comment by 130s** [comment]
ROS Melodic release seems to be set in stone [on May 23rd](https://discourse.ros.org/t/ros-melodic-morenia-beta-has-begun/4743/2?u=130s) and I assume new release needs to be made prior to that date ([asked exact hold day](https://discourse.ros.org/t/ros-melodic-morenia-beta-has-begun/4743/2?u=130s)). I'll try my best to make the official release this time (would it be the first time for MoveIt! to make it?), so let me know anything holds up `bloom`ing.

**Comment by IanTheEngineer** [comment]
I would advocate for TF2 Migration https://github.com/ros-planning/moveit/pull/830 and `boost::shared_ptr` removal https://github.com/ros-planning/moveit/pull/896 to be merged into `melodic-devel` prior to a Melodic release. They are important for ROS2 efforts, and break API. I will try to make sure they're merged soon to unblock the release.

**Comment by rhaschke** [comment]
@mamoll The release of OMPL into Melodic is missing the `ompl-config.cmake` file. This blocks source builds.
```sh
> ls /opt/ros/melodic/share/ompl/
demos  ompl.conf
```
```sh
ls /opt/ros/kinetic/share/ompl/
demos  ompl.conf  ompl-config.cmake
```

**Comment by 130s** [comment]
@IanTheEngineer I see #830 being closed :fireworks: 

@rhaschke Can I ask you to give a "go" for OMPL issue https://github.com/ros-planning/moveit/issues/806#issuecomment-390029076? I can't make a judgement for that.

There are still several unchecked items. I think the last possible day I can bloom to make into Melodic official release is probably 21th Monday (based on the [response from Melodic manager](https://discourse.ros.org/t/ros-melodic-morenia-beta-has-begun/4743/3)), which sounds a bit unrealistic. So, no pressure folks.

**Comment by rhaschke** [comment]
@130s MoveIt will not build on Melodic until OMPL is re-released by @mamoll.
For now, I added a quick hack into our Travis docker container to see remaining issues at least.
By the way, is there anybody actually **using** the Melodic branch?

**Comment by rhaschke** [comment]
@130s, @mamoll Having changed `find_package(OMPL)` -> `find_package(ompl)` as suggested [here](https://github.com/ros/catkin/issues/938#issuecomment-390285366), solves the issue for Melodic too. No need to do another release.

**Comment by mamoll** [comment]
Does that OMPL is treated as a pure cmake package and this is not necessary?:
https://github.com/ompl/ompl/blob/0f4f2ef283d5390a118d9bec3fbbe92efd0fe988/CMakeLists.txt#L245-L249
If so, can this file be deleted?:
https://github.com/ros-gbp/ompl-release/blob/master/melodic/package.xml

**Comment by 130s** [comment]
AFAIU all blocking items are checked, thanks to some of you particularly @rhaschke. If I hear no objection I'll bloom in a few hours.

**Comment by 130s** [comment]
Sorry for taking me awhile, but I bloomed https://github.com/ros/rosdistro/pull/17942

I couldn't finish organizing changelog...I've started editing changelog that was generated from `catkin_generate_changelog`, then I got called in for urgent work stuff. I'd rather just make a release with unorganized set of changelog than delaying the release. In the future release I'll try sorting them out.

**Comment by rhaschke** [comment]
> Does that OMPL is treated as a pure cmake package and this is not necessary?:
https://github.com/ompl/ompl/blob/0f4f2ef283d5390a118d9bec3fbbe92efd0fe988/CMakeLists.txt#L245-L249
If so, can this file be deleted?:
https://github.com/ros-gbp/ompl-release/blob/master/melodic/package.xml

@mamoll At least, MoveIt uses OMPL as a cmake-only package and the package.xml could be removed.

**Comment by rhaschke** [comment]
That's a valid point. But, @mamoll could decide to release OMPL through a different mechanism instead.

**Comment by mamoll** [comment]
I'll stick with bloom for now. It makes releasing packages for the different ROS versions easy. Thanks, @jspricke and @rhaschke for helping me understand what's necessary and sufficient for ROS releases ;-).

**Comment by 130s** [comment]
> - [ ]  Make an announcement

This was suggested in today's mtg.
- We should announce that MoveIt! binary is available on Melodic
- It should NOT be expected to be stable yet. Maybe calling it "beta"?

**Comment by 130s** [comment]
> - [ ]  Make an announcement

This was suggested in today's mtg.
- We should announce that MoveIt! binary is available on Melodic
- It should NOT be expected to be stable yet. Maybe calling it "beta"?
- Mention about Lunar situation https://github.com/ros-planning/moveit/issues/889

I can go down for this rabbit hole.
